### DISCUSSION TITLE
Implement custom editor for some (not all) text cells

### DISCUSSION THREAD

**Discussion by ssvendsen** [glideapps/glide-data-grid#448] [Q&A] [answered]
Hi. I have a case where I want to show a custom editor (filterable list of suggestions) for text cells in one column, while keeping the standard editor for text cells in other columns. 
I have been able to provide a custom editor for normal text cells through 'provideEditor', but with this approach it seems I can't distinguish between the cases where the built-in editor and and my custom editor should be used (in both cases the cell type is 'text').
I am thinking I have to implement a full custom cell type, but in this case - can I somehow reuse the built-in text cell renderer?

**Comment by jassmith** [comment, marked as answer]
```tsx
  provideEditor: () => p => {
        const { isHighlighted, onChange, value } = p;
        return (
            <TextCellEntry
                highlight={isHighlighted}
                autoFocus={true}
                value={value.data.YOUR_CUSTOM_CELL_DATA_PROP ?? ""}
                onChange={e =>
                    onChange({
                        ...value,
                        data: {
                            ...value.data,
                            name: e.target.value,
                        },
                    })
                }
            />
        );
    },
```

Thats pretty much the text-cell editor code. We often do this in our own custom cells.

**Reply by jassmith** [reply]
TextCellEntry is exported from `@glideapps/glide-data-grid`
